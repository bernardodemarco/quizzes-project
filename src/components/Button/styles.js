import styled, { css } from 'styled-components';

export const Button = styled.button`
  ${({ theme, background, fontSize, fontWeight, lineHeight }) => css`
    background-color: ${background ? theme.colors.primary : theme.colors.white};
    color: ${background ? theme.colors.white : theme.colors.dark};
    font-size: ${theme.fonts.fontSizes.buttons[fontSize]};
    font-weight: ${theme.fonts.fontWeights[fontWeight]};
    line-height: ${lineHeight};

    font-family: inherit;
    width: 100%;
    height: 5.6rem;
    text-align: center;
    border: none;
    border-radius: 1.6rem;
    padding: 1.6rem 3.2rem;

    &:hover {
      cursor: pointer;
    }
  `}
`;
