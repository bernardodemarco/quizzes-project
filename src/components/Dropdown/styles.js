import styled, { css } from 'styled-components';

export const Dropdown = styled.ul`
  ${({ theme, isVisible }) => css`
    visibility: hidden;
    opacity: 0;

    transition: all 300ms ease-in-out;

    ${
      isVisible &&
      css`
      visibility: visible;
      opacity: 1;
    `
    }

    position: absolute;
    top: 10rem;
    right: 38%;

    width: 14.5rem;
    height: 15.2rem;

    background-color: ${theme.colors.white};
    border-radius: 0.8rem;
    box-shadow: 0rem 0.4rem 0.4rem rgba(0, 0, 0, 0.25);

    list-style: none;

    @media (max-width: 785px) {
      visibility: hidden;
      opacity: 0;
    }
  `}
`;

export const DropdownItem = styled.li`
  ${({ theme }) => css`
    width: 100%;

    padding: 1rem 4.5rem;

    border-bottom: 0.1rem solid ${theme.colors.gray};

    &:last-child {
      border-bottom: none;
    }

    &:hover {
      cursor: pointer;
    }
  `}
`;

export const Text = styled.p`
  ${({ theme }) => css`
    color: #000000;

    text-align: center;

    font-size: ${theme.fonts.fontSizes.paragraphs.large};
    font-weight: ${theme.fonts.fontWeights.normal};
    line-height: 1.8rem;
  `}
`;
