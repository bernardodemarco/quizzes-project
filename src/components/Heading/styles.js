import styled, { css } from 'styled-components';

export const Heading = styled.h1`
  ${({ theme, fontSize, fontWeight, lineHeight }) => css`
    color: ${theme.colors.dark};
    font-size: ${theme.fonts.fontSizes.headings[fontSize]};
    font-weight: ${theme.fonts.fontWeights[fontWeight]};
    line-height: ${lineHeight};
  `}
`;
