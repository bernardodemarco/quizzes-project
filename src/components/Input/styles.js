import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    border: 0.1rem solid ${theme.colors.gray};
    border-radius: 1.2rem;

    width: 100%;
    height: 5.3rem;
    padding: 1.6rem;
    display: flex;
    justify-content: space-between;
    align-items: center;
  `}
`;

export const Input = styled.input`
  ${({ theme }) => css`
    width: 75%;
    border: none;
    font-family: inherit;
    font-weight: ${theme.fonts.fontWeights.light};
    font-size: ${theme.fonts.fontSizes.paragraphs.large};
    color: ${theme.colors.darkGray};
  `}
`;

export const Icon = styled.img`
  ${({ hover = false }) => css`
    width: 3rem;
    height: 2.5rem;

    &:hover {
      cursor: ${hover ? 'pointer' : 'default'};
    }
  `}
`;
