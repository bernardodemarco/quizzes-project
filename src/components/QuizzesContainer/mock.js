export const notAnsweredQuizzes = [
  {
    id: '224a68d0-d49a-4992-8b00-e93c0960d196',
    title: 'UI',
    search: 'ui',
    shortDescription: 'Um quiz sobre UI que vai mudar sua vida!',
    imgSrc:
      'https://sujeitoprogramador.com/wp-content/uploads/2020/12/uiDesign-1.png',
    numOfQuestions: 3,
    difficulty: 'easy',
    isAnswered: false,
    answeredDate: '2022-06-21 22:32',
    numOfCorrectAnswers: 1,
  },
  {
    id: '95b93857-476d-450d-b9a7-0ef0bd6c2cc5',
    title: 'Conceitos básicos de HTML',
    search: 'html',
    shortDescription: 'Um quiz sobre HTML que vai mudar sua vida!',
    imgSrc:
      'https://www.hostinger.com.br/tutoriais/wp-content/uploads/sites/12/2018/07/Como-inserir-imagem-ou-logo-usando-HTML-no-seu-site-1.png',
    numOfQuestions: 4,
    difficulty: 'easy',
    isAnswered: false,
    answeredDate: null,
    numOfCorrectAnswers: null,
  },
  {
    id: 'e3a8c1c8-cdc8-4939-b2ac-ad682707937d',
    title: 'HTML avançado',
    search: 'html',
    shortDescription: 'Conceitos avançados de HTML',
    imgSrc:
      'https://marquesfernandes.com/wp-content/uploads/2020/04/florian-olivo-4hbJ-eymZ1o-unsplash.jpg',
    numOfQuestions: 4,
    difficulty: 'hard',
    isAnswered: false,
    answeredDate: null,
    numOfCorrectAnswers: null,
  },
  {
    id: '224a68d0-d49a-4992-8b00-e93c0960d1961',
    title: 'UI',
    search: 'ui',
    shortDescription: 'Um quiz sobre UI que vai mudar sua vida!',
    imgSrc:
      'https://sujeitoprogramador.com/wp-content/uploads/2020/12/uiDesign-1.png',
    numOfQuestions: 3,
    difficulty: 'easy',
    isAnswered: false,
    answeredDate: '2022-06-21 22:32',
    numOfCorrectAnswers: 1,
  },
  {
    id: '95b93857-476d-450d-b9a7-0ef0bd6c2cc51',
    title: 'Conceitos básicos de HTML',
    search: 'html',
    shortDescription: 'Um quiz sobre HTML que vai mudar sua vida!',
    imgSrc:
      'https://www.hostinger.com.br/tutoriais/wp-content/uploads/sites/12/2018/07/Como-inserir-imagem-ou-logo-usando-HTML-no-seu-site-1.png',
    numOfQuestions: 4,
    difficulty: 'easy',
    isAnswered: false,
    answeredDate: null,
    numOfCorrectAnswers: null,
  },
  {
    id: 'e3a8c1c8-cdc8-4939-b2ac-ad682707937d1',
    title: 'HTML avançado',
    search: 'html',
    shortDescription: 'Conceitos avançados de HTML',
    imgSrc:
      'https://marquesfernandes.com/wp-content/uploads/2020/04/florian-olivo-4hbJ-eymZ1o-unsplash.jpg',
    numOfQuestions: 4,
    difficulty: 'hard',
    isAnswered: false,
    answeredDate: null,
    numOfCorrectAnswers: null,
  },
  {
    id: '224a68d0-d49a-4992-8b00-e93c0960d1962',
    title: 'UI',
    search: 'ui',
    shortDescription: 'Um quiz sobre UI que vai mudar sua vida!',
    imgSrc:
      'https://sujeitoprogramador.com/wp-content/uploads/2020/12/uiDesign-1.png',
    numOfQuestions: 3,
    difficulty: 'easy',
    isAnswered: false,
    answeredDate: '2022-06-21 22:32',
    numOfCorrectAnswers: 1,
  },
  {
    id: '95b93857-476d-450d-b9a7-0ef0bd6c2cc52',
    title: 'Conceitos básicos de HTML',
    search: 'html',
    shortDescription: 'Um quiz sobre HTML que vai mudar sua vida!',
    imgSrc:
      'https://www.hostinger.com.br/tutoriais/wp-content/uploads/sites/12/2018/07/Como-inserir-imagem-ou-logo-usando-HTML-no-seu-site-1.png',
    numOfQuestions: 4,
    difficulty: 'easy',
    isAnswered: false,
    answeredDate: null,
    numOfCorrectAnswers: null,
  },
  {
    id: 'e3a8c1c8-cdc8-4939-b2ac-ad682707937d2',
    title: 'HTML avançado',
    search: 'html',
    shortDescription: 'Conceitos avançados de HTML',
    imgSrc:
      'https://marquesfernandes.com/wp-content/uploads/2020/04/florian-olivo-4hbJ-eymZ1o-unsplash.jpg',
    numOfQuestions: 4,
    difficulty: 'hard',
    isAnswered: false,
    answeredDate: null,
    numOfCorrectAnswers: null,
  },
];

export const answeredQuizzes = [
  {
    id: '224a68d0-d49a-4992-8b00-e93c0960d196',
    title: 'UI',
    search: 'ui',
    shortDescription: 'Um quiz sobre UI que vai mudar sua vida!',
    imgSrc:
      'https://sujeitoprogramador.com/wp-content/uploads/2020/12/uiDesign-1.png',
    numOfQuestions: 3,
    difficulty: 'easy',
    isAnswered: true,
    answeredDate: '2022-06-21 22:32',
    numOfCorrectAnswers: 1,
  },
  {
    id: '224a68d0-d49a-4992-8b00-e93c0960d1961',
    title: 'CSS',
    search: 'ui',
    shortDescription: 'Um quiz sobre CSS que vai mudar sua vida!',
    imgSrc:
      'https://sujeitoprogramador.com/wp-content/uploads/2020/12/uiDesign-1.png',
    numOfQuestions: 10,
    difficulty: 'easy',
    isAnswered: true,
    answeredDate: '2022-06-21 22:32',
    numOfCorrectAnswers: 8,
  },
  {
    id: '224a68d0-d49a-4992-8b00-e93c0960d1962',
    title: 'React.js',
    search: 'ui',
    shortDescription: 'Um quiz sobre React.js que vai mudar sua vida!',
    imgSrc:
      'https://sujeitoprogramador.com/wp-content/uploads/2020/12/uiDesign-1.png',
    numOfQuestions: 8,
    difficulty: 'hard',
    isAnswered: true,
    answeredDate: '2022-06-21 22:32',
    numOfCorrectAnswers: 3,
  },
  {
    id: '224a68d0-d49a-4992-8b00-e93c0960d1963',
    title: 'Vue.js',
    search: 'ui',
    shortDescription: 'Um quiz sobre Vue.js que vai mudar sua vida!',
    imgSrc:
      'https://sujeitoprogramador.com/wp-content/uploads/2020/12/uiDesign-1.png',
    numOfQuestions: 12,
    difficulty: 'hard',
    isAnswered: true,
    answeredDate: '2022-06-21 22:32',
    numOfCorrectAnswers: 6,
  },
  {
    id: '224a68d0-d49a-4992-8b00-e93c0960d1946',
    title: 'Python',
    search: 'ui',
    shortDescription: 'Um quiz sobre Python que vai mudar sua vida!',
    imgSrc:
      'https://sujeitoprogramador.com/wp-content/uploads/2020/12/uiDesign-1.png',
    numOfQuestions: 3,
    difficulty: 'easy',
    isAnswered: true,
    answeredDate: '2022-06-21 22:32',
    numOfCorrectAnswers: 3,
  },
];
