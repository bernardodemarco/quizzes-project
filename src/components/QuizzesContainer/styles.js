import styled, { css } from 'styled-components';

export const Grid = styled.div`
  ${({ theme, isAnswered, numOfResults }) => css`
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 4rem;

    ${
      isAnswered &&
      css`
        margin-top: 16.8rem;
        margin-bottom: 4.8rem;
    `
    }

    ${
      !isAnswered &&
      numOfResults !== null &&
      css`
        margin-top: 2.5rem;
      `
    }

    @media ${theme.media.largeDevices} {
      gap: 2rem;
    }

    @media ${theme.media.mediumDevices} {
      grid-template-columns: 1fr;
      gap: 4rem;
    }

    ${
      !isAnswered &&
      css`
        @media (max-width: 785px) {
          margin-top: 5rem;
          margin-bottom: 5rem;
        }
      `
    }

  `}
`;
