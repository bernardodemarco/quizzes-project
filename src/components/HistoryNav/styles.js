import styled, { css } from 'styled-components';

import { Container as PageContainer } from './../PageContainer/styles';
import { ImageContainer as ReturnButton } from './../ReturnButton/styles';
import { Heading } from './../Heading/styles';

export const Nav = styled.nav`
  ${({ theme }) => css`
    z-index: 1;
    position: fixed;
    top: 0;
    right: 0;
    left: 0;

    height: 12rem;
    border-bottom: 0.1rem solid ${theme.colors.gray};

    ${PageContainer} {
      min-height: initial;
      height: 100%;
      justify-content: flex-start;
      align-items: center;
    }

    ${Heading} {
      z-index: 1;
      position: fixed;
      left: 14.7rem;
    }

    @media ${theme.media.mediumDevices} {
      ${ReturnButton} {
        left: 5rem;
      }
    }

    @media ${theme.media.smallDevices} {
      ${Heading} {
        left: 12rem;
        font-size: ${theme.fonts.fontSizes.headings.medium};
      }
    }
  `}
`;
