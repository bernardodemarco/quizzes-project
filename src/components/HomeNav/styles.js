import styled, { css } from 'styled-components';

import { Container as PageContainer } from '../PageContainer/styles';
import { Input, Container as InputContainer } from '../Input/styles';
import { Heading } from '../Heading/styles';

export const Button = styled.button`
  ${({ theme }) => css`
    display: none;

    z-index: 2;
    position: fixed;
    top: 3rem;
    right: 3rem;

    height: 4rem;
    width: 4rem;

    color: ${theme.colors.primary};
    background-color: ${theme.colors.lightGray};
    border: none;

    &:hover {
      cursor: pointer;
    }

    @media (max-width: 785px) {
      display: block;
    }
  `}
`;

export const Container = styled.div`
  ${({ theme, isVisible }) => css`
    height: 12rem;

    z-index: 1;
    position: fixed;
    top: 0;
    left: 0;
    right: 0;

    border-bottom: 0.1rem solid ${theme.colors.gray};

    transition: all 300ms ease-in-out;

    & ${PageContainer} {
      min-height: initial;
      height: 100%;
    }

    @media (max-width: 1300px) {
      & ${PageContainer} {
        padding: 0 3rem;
      }
    }

    @media (max-width: 785px) {
      & {
        height: 100vh;
        bottom: 0;
        border-bottom: none;

        visibility: hidden;
        opacity: 0;
        ${
          isVisible &&
          css`
          visibility: visible;
          opacity: 1;
        `
        }
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${() => css`
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: flex-end;

    @media (max-width: 785px) {
      & {
        height: 100%;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        gap: 7.5rem;
      }
    }
  `}
`;

export const Logo = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    align-items: flex-start;

    & ${Heading} {
      text-align: center;
    }

    @media (max-width: 1130px) {
      & ${Heading} {
        font-size: ${theme.fonts.fontSizes.headings.medium};
        margin-right: 2rem;
      }
    }

    @media ${theme.media.mediumDevices} {
      & ${Heading} {
        font-size: ${theme.fonts.fontSizes.headings.small};
      }
    }

    @media (max-width: 785px) {
      & {
        justify-content: center;
        align-items: center;
      }

      & ${Heading} {
        margin-right: 0;
      }
    }
  `}
`;

export const Nav = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-direction: row;
    align-items: flex-end;
    gap: 3rem;

    & ${Input} {
      width: 40.4rem;
    }

    @media (max-width: 1130px) {
      & {
        align-items: center;
      }

      & ${Input} {
        width: 30rem;
      }
    }

    @media ${theme.media.mediumDevices} {
      & ${Input} {
        width: 22.5rem;
      }
    }

    @media (max-width: 785px) {
      & {
        flex-direction: column;
        justify-content: center;
        align-items: center;
      }

      & ${InputContainer}, & ${Input}, & .dropdown-button  {
        display: none;
      }
    }
  `}
`;

export const Link = styled.a`
  ${({ theme }) => css`
    display: block;
    text-align: center;
    color: ${theme.colors.dark};

    font-weight: ${theme.fonts.fontWeights.light};
    font-size: ${theme.fonts.fontSizes.paragraphs.large};
    line-height: 2.1rem;

    &:hover {
      cursor: pointer;
    }

    @media (max-width: 785px) {
      & {
        font-size: 1.8rem;
      }
    }
  `}
`;

export const Theme = styled.p`
  ${({ theme }) => css`
    display: none;
    color: ${theme.colors.dark};
    text-align: center;

    font-weight: ${theme.fonts.fontWeights.light};
    font-size: ${theme.fonts.fontSizes.paragraphs.large};
    line-height: 2.1rem;

    &:hover {
      cursor: pointer;
    }

    @media (max-width: 785px) {
      & {
        display: block;
        font-size: 1.8rem;
      }
    }
  `}
`;
