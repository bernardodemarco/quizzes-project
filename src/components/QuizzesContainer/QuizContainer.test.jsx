import { QuizzesContainer } from '.';
import { notAnsweredQuizzes, answeredQuizzes } from './mock';

import { BrowserRouter as Router } from 'react-router-dom';
import { renderThemeProvider } from './../../styles/render-theme-provider';
import { fireEvent, screen } from '@testing-library/react';

describe('<QuizzesContainer />', () => {
  it('should render <QuizzesContainer /> with unanswered quizzes correctly', () => {
    renderThemeProvider(
      <Router>
        <QuizzesContainer quizzes={notAnsweredQuizzes} />
      </Router>,
    );
    expect(screen.getAllByRole('heading')).toHaveLength(
      notAnsweredQuizzes.length,
    );
    fireEvent.click(screen.getAllByRole('heading')[0]);
  });

  it('should render <QuizzesContainer /> with answered quizzes correctly', () => {
    renderThemeProvider(
      <Router>
        <QuizzesContainer quizzes={answeredQuizzes} />
      </Router>,
    );
    expect(screen.getAllByRole('heading')).toHaveLength(answeredQuizzes.length);
    fireEvent.click(screen.getAllByRole('heading')[0]);
  });

  it('should match snapshot', () => {
    const { container } = renderThemeProvider(
      <Router>
        <QuizzesContainer quizzes={notAnsweredQuizzes} />
      </Router>,
    );
    expect(container).toMatchSnapshot();
  });
});
