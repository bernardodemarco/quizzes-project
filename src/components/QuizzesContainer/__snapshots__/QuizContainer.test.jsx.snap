// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuizzesContainer /> should match snapshot 1`] = `
.c1 {
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 4rem;
  margin-top: 2.5rem;
}

.c2 {
  overflow: hidden;
  border: 0.1rem solid #BEBAB3;
  border-radius: 0.8rem;
  -webkit-filter: drop-shadow(0px 4px 4px rgba(0,0,0,0.25));
  filter: drop-shadow(0px 4px 4px rgba(0,0,0,0.25));
  width: 55.2rem;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  background-color: #FFFFFF;
}

.c2:hover {
  cursor: pointer;
}

.c3 {
  position: relative;
  width: 100%;
  height: 19.4rem;
  background-image: url(https://sujeitoprogramador.com/wp-content/uploads/2020/12/uiDesign-1.png);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
}

.c8 {
  position: relative;
  width: 100%;
  height: 19.4rem;
  background-image: url(https://www.hostinger.com.br/tutoriais/wp-content/uploads/sites/12/2018/07/Como-inserir-imagem-ou-logo-usando-HTML-no-seu-site-1.png);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
}

.c9 {
  position: relative;
  width: 100%;
  height: 19.4rem;
  background-image: url(https://marquesfernandes.com/wp-content/uploads/2020/04/florian-olivo-4hbJ-eymZ1o-unsplash.jpg);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
}

.c5 {
  width: 100%;
  padding: 1.6rem 1.6rem 0.8rem 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  gap: 0.4rem;
}

.c4 {
  position: absolute;
  bottom: 0.8rem;
  right: 1.6rem;
  padding: 0.4rem 1.6rem;
  border-radius: 1.2rem;
  color: #F2F2F2;
  background-color: #65AAEA;
  text-transform: uppercase;
  font-size: 1.4rem;
  font-weight: 500;
  line-height: 1.6rem;
}

.c10 {
  position: absolute;
  bottom: 0.8rem;
  right: 1.6rem;
  padding: 0.4rem 1.6rem;
  border-radius: 1.2rem;
  color: #F2F2F2;
  background-color: #EF4949;
  text-transform: uppercase;
  font-size: 1.4rem;
  font-weight: 500;
  line-height: 1.6rem;
}

.c7 {
  font-size: 1.4rem;
  font-weight: 400;
  color: #3C3A36;
  text-align: center;
  line-height: 2.1rem;
}

.c0 {
  color: #3C3A36;
  font-size: 2rem;
  font-weight: 500;
  line-height: 26px;
}

.c6 {
  color: #3C3A36;
  font-size: 2.4rem;
  font-weight: 500;
  line-height: normal;
}

@media (max-width:1200px) {
  .c1 {
    gap: 2rem;
  }
}

@media (max-width:974px) {
  .c1 {
    grid-template-columns: 1fr;
    gap: 4rem;
  }
}

@media (max-width:785px) {
  .c1 {
    margin-top: 5rem;
    margin-bottom: 5rem;
  }
}

@media (max-width:1200px) {
  .c2 {
    width: 45rem;
  }
}

@media (max-width:974px) {
  .c2 {
    width: 55.2rem;
  }
}

@media (max-width:575px) {
  .c2 {
    width: 90vw;
  }

  .c2 p {
    text-align: start;
  }
}

<div>
  <h2
    class="c0"
    font-size="xSmall"
    font-weight="normal"
  >
     Resultados
  </h2>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          easy
        </span>
      </div>
      <div
        class="c5"
      >
        <h3
          class="c6"
          font-size="small"
          font-weight="normal"
        >
          UI
        </h3>
        <p
          class="c7"
          color="dark"
          font-size="medium"
          font-weight="light"
        >
          Um quiz sobre UI que vai mudar sua vida!
        </p>
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c8"
      >
        <span
          class="c4"
        >
          easy
        </span>
      </div>
      <div
        class="c5"
      >
        <h3
          class="c6"
          font-size="small"
          font-weight="normal"
        >
          Conceitos básicos de HTML
        </h3>
        <p
          class="c7"
          color="dark"
          font-size="medium"
          font-weight="light"
        >
          Um quiz sobre HTML que vai mudar sua vida!
        </p>
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c9"
      >
        <span
          class="c10"
        >
          hard
        </span>
      </div>
      <div
        class="c5"
      >
        <h3
          class="c6"
          font-size="small"
          font-weight="normal"
        >
          HTML avançado
        </h3>
        <p
          class="c7"
          color="dark"
          font-size="medium"
          font-weight="light"
        >
          Conceitos avançados de HTML
        </p>
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          easy
        </span>
      </div>
      <div
        class="c5"
      >
        <h3
          class="c6"
          font-size="small"
          font-weight="normal"
        >
          UI
        </h3>
        <p
          class="c7"
          color="dark"
          font-size="medium"
          font-weight="light"
        >
          Um quiz sobre UI que vai mudar sua vida!
        </p>
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c8"
      >
        <span
          class="c4"
        >
          easy
        </span>
      </div>
      <div
        class="c5"
      >
        <h3
          class="c6"
          font-size="small"
          font-weight="normal"
        >
          Conceitos básicos de HTML
        </h3>
        <p
          class="c7"
          color="dark"
          font-size="medium"
          font-weight="light"
        >
          Um quiz sobre HTML que vai mudar sua vida!
        </p>
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c9"
      >
        <span
          class="c10"
        >
          hard
        </span>
      </div>
      <div
        class="c5"
      >
        <h3
          class="c6"
          font-size="small"
          font-weight="normal"
        >
          HTML avançado
        </h3>
        <p
          class="c7"
          color="dark"
          font-size="medium"
          font-weight="light"
        >
          Conceitos avançados de HTML
        </p>
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <span
          class="c4"
        >
          easy
        </span>
      </div>
      <div
        class="c5"
      >
        <h3
          class="c6"
          font-size="small"
          font-weight="normal"
        >
          UI
        </h3>
        <p
          class="c7"
          color="dark"
          font-size="medium"
          font-weight="light"
        >
          Um quiz sobre UI que vai mudar sua vida!
        </p>
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c8"
      >
        <span
          class="c4"
        >
          easy
        </span>
      </div>
      <div
        class="c5"
      >
        <h3
          class="c6"
          font-size="small"
          font-weight="normal"
        >
          Conceitos básicos de HTML
        </h3>
        <p
          class="c7"
          color="dark"
          font-size="medium"
          font-weight="light"
        >
          Um quiz sobre HTML que vai mudar sua vida!
        </p>
      </div>
    </div>
    <div
      class="c2"
    >
      <div
        class="c9"
      >
        <span
          class="c10"
        >
          hard
        </span>
      </div>
      <div
        class="c5"
      >
        <h3
          class="c6"
          font-size="small"
          font-weight="normal"
        >
          HTML avançado
        </h3>
        <p
          class="c7"
          color="dark"
          font-size="medium"
          font-weight="light"
        >
          Conceitos avançados de HTML
        </p>
      </div>
    </div>
  </div>
</div>
`;
